### PR TITLE
now ignores non-200 HTTP status codes

### PR DESCRIPTION
**Pull request by Jtcollins90**
Set func queryEndpoint to ignore any non 200 HTTP status codes to avoid errors.

### CODE REVIEW

File: src/fullerite/collector/nerve_uwsgi.go [hunk 1]
```diff
@@ -304,6 +304,11 @@ func queryEndpoint(endpoint string, timeout int) ([]byte, string, error) {
 	}
 
 	rsp, err := client.Get(endpoint)
+	//TODO: Insert check for 4xx
+	if (err == nil) && (rsp.StatusCode != 200) {
+		return []byte{}, "Non 200 Status Code. ", err
+	}
```
Comment (gnufied) [inline src/fullerite/collector/nerve_uwsgi.go]: Few problems:

1. second return parameter is metrics-schema version, so if nothing is available we should send empty string.
2. if `err` is nil (which will be in this case)  - whoever called this method is still going to think that `http` call succeeded and try to extract metrics where there is none available.

**Comment by Jtcollins90** [comment]
Amended edge case so that the metrics-scema return parameter is now an empty string and the error output is no longer nil.

**Comment by gnufied** [comment]
:shipit:

File: src/fullerite/collector/nerve_uwsgi.go [hunk 2]
```diff
@@ -304,6 +305,12 @@ func queryEndpoint(endpoint string, timeout int) ([]byte, string, error) {
 	}
 
 	rsp, err := client.Get(endpoint)
+
+	if (err == nil) && (rsp.StatusCode != 200) {
```
Comment (gnufied) [inline src/fullerite/collector/nerve_uwsgi.go]: while this works - it is more cleaner to write:

```
if rsp && rsp.StatusCode != 200 {
}
```

and get rid of extra parenthesis etc.  It is more idiomatic.

**Pull request merged**